### PR TITLE
OSGI

### PR DESCRIPTION
**Pull request by jenniferboedker**
**Purpose of change**
Make the library OSGI ready to be included in the Liferay 7 version. Core-Utils-Lib should be deployed as OSGI dependency and still be included into stand-alone cli applications.

**Changes**
Add profiles to the POM so that the dependencies can be set as provided for the OSGI version and set as compile for stand-alone cli applications. Furthermore, the dependencies itself may need to be OSGI-fied.

### CODE REVIEW

**Comment by jenniferboedker** [comment]
[port core-utils-lib to OSGI](https://trello.com/c/9YFfeahb/44-port-core-utils-lib-to-osgi)

File: bnd.bnd [hunk 1]
```diff
@@ -1,8 +1,7 @@
...
 Bundle-Version: ${project.version}
 Bundle-SymbolicName: ${project.groupId}.${project.artifactId}
-Bundle-DocURL: ${project.url}
-
-Bundle-Activator: life.qbic.example.Activator
-
-Export-Package: life.qbic.example
+Bundle-URL: ${project.url}
+Export-Package: life.qbic.*;version="${project.version}"
+Import-Package: !org.everit.json.schema.*,!org.json.*,!org.xml.sax,\\
```
Comment (jenniferboedker) [inline bnd.bnd:6]: should this be two backslashes? I would expect one

File: src/main/groovy/life/qbic/Main.groovy [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import life.qbic.services.ServiceConnector
+import life.qbic.services.ServiceType
+import life.qbic.services.connectors.ConsulConnector
+
+/**
+ * <class short description - 1 Line!>
+ *
+ * <More detailed description - When to use, what it solves, etc.>
+ *
+ * @since <versiontag>
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/Main.groovy:14]: Missing description

File: src/main/groovy/life/qbic/Main.groovy [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public static void main(String[] args) {
+        URL registryUrl = new URL("http://service-registry-test.am10.uni-tuebingen" +
+                ".de:8080/registry/v1")
+        ServiceConnector consulConnector = new ConsulConnector(registryUrl)
+        List< Service> services = new ConsulServiceFactory(consulConnector)
+                .getServicesOfType(ServiceType.SAMPLE_TRACKING)
+        services.each {
+            println it.rootUrl
+        }
+    }
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/Main.groovy:27]: I assume this is for testing purposes to see if removing micronaut worked?

**Review by jenniferboedker — COMMENTED** [review]
Nice work, I will try it out with my local liferay instance and give you feedback. Other than that I left some comments for things that are not clear to me

File: bnd.bnd (hunk 1 above)
Comment (sven1103) [reply bnd.bnd:6]: That's the syntax that worked to successfully create the bundle :)

File: src/main/groovy/life/qbic/Main.groovy (hunk 2 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/Main.groovy:14]: 🤦

File: src/main/groovy/life/qbic/Main.groovy (hunk 3 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/Main.groovy:27]: Jup

**Review by jenniferboedker — COMMENTED** [review]
LGTM but can't approve since I am the author of this PR 😅

File: pom.xml [hunk 4]
```diff
@@ -84,27 +84,37 @@
...
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.12.0</version>
+            <scope>${osgi.scope}</scope>
         </dependency>
         <dependency>
             <artifactId>org.everit.json.schema</artifactId>
             <groupId>com.github.everit-org.json-schema</groupId>
             <version>1.12.1</version>
+            <scope>compile</scope>
```
Comment (Steffengreiner) [inline pom.xml:117]: why is this dependency set as compile while other external dependencies are set as osgi.scope?

**Review by Steffengreiner — APPROVED** [review]
LGTM!

**Review by sven1103 — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (sven1103) [reply pom.xml:117]: because we always need it it to be available to the classpath during build time, and it is not provided by the OSGi framework

**Pull request merged**